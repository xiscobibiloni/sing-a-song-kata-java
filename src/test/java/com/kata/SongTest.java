package com.kata;

import static com.github.stefanbirkner.systemlambda.SystemLambda.tapSystemOut;
import org.approvaltests.Approvals;
import org.junit.jupiter.api.DisplayNameGeneration;
import org.junit.jupiter.api.DisplayNameGenerator.ReplaceUnderscores;
import org.junit.jupiter.api.Test;


@DisplayNameGeneration(ReplaceUnderscores.class)
class SongTest {

    @Test
    void sing_a_song() throws Exception {

        String song = tapSystemOut(() -> {
            Song.main(new String[]{"fly", "spider", "bird", "cat", "dog", "cow", "horse"});
        });

        Approvals.verify(song);
    }

    @Test
    void sing_a_song_with_minus_animals() throws Exception {

        String song = tapSystemOut(() -> {
            Song.main(new String[]{"cow", "cat", "spider", "fly", "horse"});
        });

        Approvals.verify(song);
    }

}
