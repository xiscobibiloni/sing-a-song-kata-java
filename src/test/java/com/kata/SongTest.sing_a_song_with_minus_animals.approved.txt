There was an old lady who swallowed a cow.
I don't know how she swallowed a cow!

There was an old lady who swallowed a cat;
Fancy that to swallow a cat!
She swallowed the cat to catch the cow;
I don't know how she swallowed a cow!

There was an old lady who swallowed a spider;
That wriggled and wiggled and tickled inside her.
She swallowed the spider to catch the cat,
She swallowed the cat to catch the cow;
I don't know how she swallowed a cow!

There was an old lady who swallowed a fly;
I don't know why she swallowed a fly - perhaps she'll die!
She swallowed the fly to catch the spider,
She swallowed the spider to catch the cat,
She swallowed the cat to catch the cow;
I don't know how she swallowed a cow!

There was an old lady who swallowed a horse...
...She's dead, of course!
